ClLike/cl_like/power_spectrum.py

```diff
@@ -119,18 +119,31 @@ def get_Pk(self):
         return self._current_state['Pk']
 
     def _get_pk_data(self, cosmo):
-        # cosmo.compute_nonlin_power()
-        # pkmm = cosmo.get_nonlin_power(name='delta_matter:delta_matter')
         pkmm = None
         if self.bias_model == 'Linear':
             cosmo.compute_nonlin_power()
             pkmm = cosmo.get_nonlin_power(name='delta_matter:delta_matter')
+            if 'delta_matter:Weyl' in cosmo._pk_nl:
+                pkwm = pkmw = cosmo.get_nonlin_power(name='delta_matter:Weyl')
+            else:
+                pkwm = pkmw = pkmm
+            if 'Weyl:Weyl' in cosmo._pk_nl:
+                pkww = cosmo.get_nonlin_power(name='Weyl:Weyl')
+            else:
+                pkww = pkmm
             pkd = {}
             pkd['pk_mm'] = pkmm
             pkd['pk_md1'] = pkmm
             pkd['pk_d1m'] = pkmm
             pkd['pk_d1d1'] = pkmm
+            pkd['pk_d1w'] = pkd['pk_wd1'] = pkwm
+            pkd['pk_mw'] = pkd['pk_wm'] = pkwm
+            pkd['pk_ww'] = pkww
         elif self.is_PT_bias:
+            if ('delta_matter:Weyl' in cosmo._pk_nl) or \
+                    ('Weyl:Weyl' in cosmo._pk_nl):
+                raise RuntimeError('Pk involving the Weyl potential not '
+                                   'implemented for PT_bias')
```

Comment:
I can add a flag to use matter if needed

Reply:
I was originally just changing `w` to `m` in the PT cases, but I think it's better that these kind of changes do not happen automatically

Reply:
but my question is: if we merge this, then we cannot use PT biases or HEFT anymore at all, right? Because anything that uses lensing will always look for `w`

Reply:
That is correct, I think. I'll change that.

Reply:
Wait, that will only be raised if the CCL cosmology object has the pk of the Weyl, which is not true in general. I need to have a look again at the code because I might have already thought about this and might not there be a problem

Reply:
Actually, in line 142 pk_ww = pk_mm if Weyl is not defined in the CCL cosmology, so you're passing the matter pk to bacco, etc.

Reply:
Ok. What I said above was wrong but I have done a few modifications for it to work. I have also fixed the tests.